core/src/main/scala/kafka/raft/KafkaMetadataLog.scala

```diff
@@ -501,6 +524,24 @@ final class KafkaMetadataLog private (
     }
   }
 
+  private def removeSnapshots(
+    expiredSnapshots: mutable.TreeMap[OffsetAndEpoch, Option[FileRawSnapshotReader]],
+    reason: SnapshotDeletionReason,
+  ): Unit = {
+    expiredSnapshots.foreach { case (snapshotId, _) =>
+      reason.logReason(snapshotId)
+      Snapshots.markForDelete(log.dir.toPath, snapshotId)
+    }
+
+    if (expiredSnapshots.nonEmpty) {
+      scheduler.scheduleOnce(
+        "delete-snapshot-files",
+        () => KafkaMetadataLog.deleteSnapshotFiles(log.dir.toPath, expiredSnapshots, this),
+        config.fileDeleteDelayMs
+      )
+    }
+  }
```

Comment:
This looks like a duplicate of the method `removeSnapshots(mutable.TreeMap[_, _])`.